build.wake

```diff
@@ -0,0 +1,430 @@
+#
+# Utility Functions
+#
+
+# This expects a sorted list of paths.
+# If a path is a prefix of the path that follows it, it is filtered out.
+def filterDirs sortedFiles =
+  def helper acc list = match list
+    head1, head2, tail =
+      def isPrefixOf path = matches (regExpCat (path.quote, `(/.+)?`, Nil))
+      if isPrefixOf head1 head2
+      then helper acc (head2, tail)
+      else helper (head1, acc) (head2, tail)
+    _ = list ++ acc
+  helper Nil sortedFiles
+  | reverse
+
+def getDestDirs srcDir destDir inputs =
+  def srcDirRegex = regExpCat (`^`, "{srcDir}/..".simplify.quote, `/+`, Nil)
+  inputs
+  | mapFlat (
+    _
+    | getPathName
+    | simplify
+    | replace srcDirRegex ""
+    | extract `(.*/)[^/]*`
+    | map ("{destDir}/{_}")
+  )
+  | distinctBy scmp
+
+def copyDir srcDir destDir fileInputs =
+  def dirInputs =
+    getDestDirs srcDir destDir fileInputs
+    | map mkdir
+  def inputs = dirInputs ++ fileInputs
+  def from_ = simplify "{srcDir}/.."
+  def srcs =
+    fileInputs
+    | map (
+      _
+      | getPathName
+      | from_.relative
+      | simplify
+    )
+    | filter (! matches `\.\./.*` _)
+    | sortBy (_ <~ _)
+    | filterDirs
+  def cmdline =
+    "rsync", "--inplace", "--relative",
+    "--ignore-times", srcs ++ (relative from_ destDir, Nil)
+  makePlan cmdline inputs
+  | setPlanDirectory from_
+  | runJob
+  | getJobOutputs
+
+def filterWakeAndDotFiles =
+  _
+  | filter (!matches `(.*/)*\..*` _.getPathName)
+  | filter (!matches `.*\.wake` _.getPathName)
+
+#
+# Plan Resources
+#
+
+global def defaultFreedomESDKResources =
+  "riscv-tools/2019.08.0",
+  "python/python/3.7.1",
+  Nil
+
+#
+# Virtualenv Creation
+#
+# Freedom E SDK depends on a number of Python libraries. Here we enable
+# the installation of a single, workspace-wide virtualenv which can be
+# preinstalled using either the `preinstallFreedomESDKVirtualenv` target
+# or the global `preinstall` topic. If the virtualenv has not been
+# preinstalled, it will be created on-demand when needed during the
+# creation of a `FreedomESDKTarget`.
+#
+
+global def freedomESDKBuildDir = "build/freedom-e-sdk"
+global def freedomESDKVenvDir = "{freedomESDKBuildDir}/venv"
+
+def makeFreedomESDKVirtualenvJob Unit =
+  def runDir = here
+
+  def inputs =
+    source "freedom-e-sdk/scripts/virtualenv.mk",
+    source "freedom-e-sdk/requirements.txt",
+    mkdir freedomESDKVenvDir,
+    Nil
+
+  def cmd =
+    def venvDir = relative runDir freedomESDKVenvDir
+    "bash", "-c", "%
+      export FREEDOM_E_SDK_VENV_PATH=$(pwd)/%{venvDir}
+      make -f scripts/virtualenv.mk virtualenv
+      # Remove e.g. .fuse/31337 from the paths in the virtualenv
+      find ${FREEDOM_E_SDK_VENV_PATH}/bin -type f | xargs sed -i 's%.fuse/[[:digit:]]\+/%%g'
+      # Remove e.g. .build/31337 from the paths in the virtualenv (when Wake doesn't use FUSE)
+      find ${FREEDOM_E_SDK_VENV_PATH}/bin -type f | xargs sed -i 's%.build/[[:digit:]]\+/%%g'
```

Comment:
Nothing wrong with your code here, as I fully understand the problem here, but this is just a self-remark that we need to fix this so that the FUSE runner resolves paths into the non-FUSE directory, such that the `.fuse` directory is basically invisible to the underlying processes.